gneiss/util.py

```diff
@@ -184,3 +195,81 @@ def rename_internal_nodes(tree, names=None, inplace=False):
             n.name = label
             i += 1
     return _tree
+
+
+def _intersect_of_table_metadata_tree(table, metadata, tree):
```

Comment:
This is part of refactoring the code.  After evaluating where this sort of functionality really should belong, I think it is more appropriate that it should go into `gneiss.util` since it is a utility function.